cnd/src/jsonrpc.rs

```diff
@@ -67,9 +64,20 @@ impl<T> Request<T> {
 #[serde(untagged)]
 pub enum Response<T> {
     Success { result: T },
-    Error { code: i64, message: String },
+    Error(JsonRpcError),
 }
 
+#[derive(Debug, serde::Deserialize, thiserror::Error)]
+#[error("JSON-RPC request failed with code {code}: {message}")]
+pub struct JsonRpcError {
+    code: i64,
+    message: String,
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("connection error: {0}")]
+pub struct ConnectionFailed(#[from] reqwest::Error);
```

Comment:
:mega: I don't think this is used anywhere now.

It used to be used implicitly through the `?` on line 30 but given that you are now using anyhow, the conversion to this no longer happens.
You can use `map_err` to restore this behaviour.

Reply:
In fact the dry tests now fail because of this!

I tried, but I could not see the original implicit conversion :(

Reply:
> I tried, but I could not see the original implicit conversion :(

Every `?` contains a `ReturnTypeError::from(inner_error)`.
This always works because of the blanket impl of `From<T>` for every `T` that just returns `self`.

Reply:
I'm still confused about one thing. Does `client.send` actually only fail with a connection error? The error enum used by `reqwest` seems to also include HTTP errors. Is this just another case of error enums being bad?

Reply:
> Is this just another case of error enums being bad?

I think so yes. "Bad" in the sense that they use just one error enum for all public APIs.